classes/announcement/AnnouncementDAO.inc.php

```diff
@@ -48,50 +50,13 @@ class AnnouncementDAO extends SchemaDAO {
 	 * @return Announcement
 	 */
 	function getById($announcementId, $assocType = null, $assocId = null) {
-		$params = array((int) $announcementId);
-		if ($assocType !== null) $params[] = (int) $assocType;
-		if ($assocId !== null) $params[] = (int) $assocId;
-		$result = $this->retrieve(
-			'SELECT	* FROM announcements WHERE announcement_id = ?' .
-			($assocType !== null?' AND assoc_type = ?':'') .
-			($assocId !== null?' AND assoc_id = ?':''),
-			$params
-		);
-
-		$returner = null;
-		if ($result->RecordCount() != 0) {
-			$returner = $this->_fromRow($result->GetRowAssoc(false));
+		$query = Capsule::table($this->tableName)->where($this->primaryKeyColumn, '=', (int) $announcementId);
+		if ($assocType !== null) $query->where('assoc_type', '=', (int) $assocType);
+		if ($assocId !== null) $query->where('assoc_id', '=', (int) $assocType);
```

Comment:
Good question; I'll test when I have a sec. Using ADODB, PostgreSQL would complain if the data type didn't match the column type; I would expect PDO to behave the same. This could cause SQL failures when binding a user-supplied parameter, if the user entered a string instead of a number. Putting the casts in the binding resulted in more not-foundish behaviour than fatal-errorish.